packages/@aws-cdk/core/README.md

```diff
@@ -815,3 +815,41 @@ const stack = new Stack(app, 'StackName', {
 ```
 
 By default, termination protection is disabled.
+
+### CfnJson
+
+`CfnJson` allows you to postpone the resolution of a JSON blob from
+deployment-time. This is useful in cases where the CloudFormation JSON template
+cannot express a certain value.
+
+A common example is to use `CfnJson` in order to render a JSON map which needs
+to use intrinsic functions in keys. Since JSON map keys must be strings, it is
+impossible to use intrinsics in keys and `CfnJson` can help.
+
+The following example defines an IAM role which can only be assumed by
+principals that are tagged with a specific tag. 
+
+```ts
+const tagParam = new CfnParameter(this, 'TagName');
+
+const stringEquals = new CfnJson(this, 'ConditionJson', {
+  value: {
+    [`aws:PrincipalTag/${tagParam.valueAsString}`]: true
+  },
+});
+
+const principal = new AccountRootPrincipal().withConditions({
+  StringEquals: stringEquals,
```

Comment:
Shouldn't are more direct representation of the example be:

```ts
const principalTag = new CfnJson(this, 'PrincipalTag', {
  value: `aws:PrincipalTag/${tagParam.valueAsString}`
}).toString();

const principal = new AccountRootPrincipal().withConditions({
  StringEquals: {
    [principalTag]: true
  }
});
```

?

In fact, this smells like it could use a helper to be slightly nicer:

```ts
const principalTag = CfnJson.asString(this, 'PrincipalTag', `aws:PrincipalTag/${tagParam.valueAsString}`);
```

Reply:
This will fail because `principalTag` resolves to a non-string `{ "Ref": "Param" }`:

```
[principalTag]: true
```

Reply:
Oh I see, that just replaces one intrinsic with another. Nevermind I'm an idiot :)